cinema/serializers.py

```diff
@@ -1 +1,84 @@
-# write serializers here
+from rest_framework import serializers
+
+from cinema.models import Movie, Genre, Actor, CinemaHall, MovieSession
+
+
+class ActorSerializer(serializers.ModelSerializer):
+    full_name = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Actor
+        fields = ("id", "first_name", "last_name", "full_name")
+
+    def get_full_name(self, obj):
+        return f"{obj.first_name} {obj.last_name}"
+
+
+class GenreSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Genre
+        fields = ("id", "name")
+
+
+class MovieSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Movie
+        fields = ("id", "title", "description", "duration", "genres", "actors")
+
+
+class MovieDetailSerializer(MovieSerializer):
+    genres = GenreSerializer(many=True, read_only=True)
+    actors = ActorSerializer(many=True, read_only=True)
+
+
+class MovieListSerializer(MovieSerializer):
+    genres = serializers.SlugRelatedField(
+        many=True, slug_field="name", read_only=True
+    )
+    actors = serializers.SerializerMethodField()
+
+    def get_actors(self, movie):
```

Comment:
if you use it this way in serializer, then most probably it will make extra db queries. You can use debug tool bar for this